File: leetcode/1342. Number of Steps to Reduce a Number to Zero/1342.cpp
// https://helloacm.com/iterative-and-recursion-algorithms-to-compute-the-number-of-steps-to-reduce-a-number-to-zero/
// https://leetcode.com/problems/number-of-steps-to-reduce-a-number-to-zero/
// EASY

class Solution {
public:
    int numberOfSteps (int num) {
        int r = 0;
        while (num != 0) {
            r ++;
            if (num % 2 == 0) {
                num >>= 1;
            } else {
                num --;
            }
        }
        return r;
    }
};

File: leetcode/1342. Number of Steps to Reduce a Number to Zero/1342.py
# https://helloacm.com/teaching-kids-programming-min-number-of-steps-to-reduce-a-number-to-zero/
# https://leetcode.com/problems/number-of-steps-to-reduce-a-number-to-zero/
# EASY, MATH

class Solution:
    def numberOfSteps(self, num: int) -> int:
        a = bin(num)[2:]
        ones = a.count('1')
        zeros = len(a) - ones
        return ones * 2 + zeros - 1
